Okta.AspNet/TokenExchanger.cs

```diff
@@ -54,6 +54,14 @@ public async Task ExchangeCodeForTokenAsync(AuthorizationCodeReceivedNotificatio
                     tokenResponse).ConfigureAwait(false);
             }
 
+            // get subject claim and duplicate it to MSFT's NameIdentifier to support the .NET MVC antiforgery provider
+            var sub = response.AuthenticationTicket.Identity.Claims.FirstOrDefault(c => c.Type == "sub")?.Value;
+            if (sub != null)
+            {
+                response.AuthenticationTicket.Identity.AddClaim(new Claim(ClaimTypes.NameIdentifier, sub));
+                response.AuthenticationTicket.Identity.AddClaim(new Claim("http://schemas.microsoft.com/accesscontrolservice/2010/07/claims/identityprovider", "ASP.NET Identity", "http://www.w3.org/2001/XMLSchema#string"));
```

Comment:
Why do you need this? Is it not sufficient having `NameIdentifier`?
Also, isn't there a constant for this `http://schemas.microsoft.com/accesscontrolservice/2010/07/claims/identityprovider`?

Reply:
The 2nd line is recommended on that link but I honestly don't know what it's for.  Looks like it works without it, so I'll take it out.